src/main/java/mate/academy/spring/controller/UserController.java

```
@@ -0,0 +1,26 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.dto.mapping.impl.response.UserResponseMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserResponseMapper responseMapper;
+
+    public UserController(UserService userService, UserResponseMapper responseMapper) {
+        this.userService = userService;
+        this.responseMapper = responseMapper;
+    }
+
+    @GetMapping(name = "/by-email")
+    public UserResponseDto getUserByEmail(@RequestParam String email) {
+        return responseMapper.toDto(userService.findByEmail(email).get());

```

let's throw an exception in case if null in optional